cpp/src/arrow/compute/api_scalar.h

```diff
@@ -30,6 +30,10 @@
 #include "arrow/util/macros.h"
 #include "arrow/util/visibility.h"
 
+#ifdef ARROW_WITH_UTF8PROC
+#include <utf8proc.h>
```

Comment:
Why are we including utf8proc here?

Reply:
You're right, we shouldn't. This is a leftover from the previous PR.